src/main/java/uk/gov/justice/digital/hmpps/keyworker/rolemigration/RoleAssignmentsService.java

```diff
@@ -32,48 +38,96 @@ public RoleAssignmentsService(RoleService roleService) {
      * Remove from the user (at the caseload) the roles identified by specification.rolesToRemove.
      */
     @PreAuthorize("hasAnyRole('MAINTAIN_ACCESS_ROLES_ADMIN')")
-    public void updateRoleAssignments(@Valid final RoleAssignmentsSpecification specification) {
+    public Map<String, RoleAssignmentStats> updateRoleAssignments(@Valid final RoleAssignmentsSpecification specification) {
         log.info("Updating role assignments: {}", specification);
+        val caseloadResults = new TreeMap<String, RoleAssignmentStats>();
         for (val caseload : specification.getCaseloads()) {
```

Comment:
would be better to do a `specification.getCaseloads().stream().map()` instead here rather than a `for` loop, even though its not great to `stream` with side effects...
To keep it simple could even add the caseload to `RoleAssignmentStats` and then you can return a `List` instead

Reply:
we have added caseload to  RoleAssignmentStats  and are now returning a list